src/main/java/com/sps/Servlets/NewRecipeServlet.java

```diff
@@ -134,15 +132,18 @@ private void addToSearchStrings(Collection<String> searchStrings, String stringT
   private Recipe entityToRecipe(Entity recipeEntity) {
     String name = (String) recipeEntity.getProperty("name");
     String description = (String) recipeEntity.getProperty("description");
-    LinkedList<String> tags = (LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("tags"), TAG);
-    LinkedList<String> ingredients = (LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("ingredients"), INGREDIENT);
+    HashSet<String> tags = new HashSet<>((LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("tags"), TAG));
```

Comment:
Why the data structure change? Jw :o

Reply:
I don't know, in the most updated version it actually becomes a LinkedHashSet. My guess is that rebasing changed this file to some older version maybe?

Reply:
If I make the change here, I think I'll have to solve merge conflicts twice for the next two PRs, so I'd prefer to leave it to the later PR.